### PR TITLE
Integrate github.com/docker/libkv

### PR DESCRIPTION
**Pull request by thaJeztah**
Note that this package will ultimately go away; this is just an intermediate step towards that; see

- https://github.com/moby/swarmkit/pull/3130
- https://github.com/moby/moby/pull/45348

A reduced set of the dependency, only taking the parts that are used. Taken from
upstream commit: https://github.com/docker/libkv/commit/dfacc563de57ee17ca54bb3b6b2a004454f3ce41

    # install filter-repo (https://github.com/newren/git-filter-repo/blob/main/INSTALL.md)
    brew install git-filter-repo
    
    cd ~/projects
    
    # create a temporary clone of docker
    git clone https://github.com/docker/libkv.git temp_libkv
    cd temp_libkv
    
    # create branch to work with
    git checkout -b migrate_libkv
    
    # remove all code, except for the files we need; rename the remaining ones to their new target location
    git filter-repo --force \
        --path libkv.go \
        --path store/store.go \
        --path store/boltdb/boltdb.go \
        --path-rename libkv.go:libnetwork/internal/kvstore/kvstore_manage.go \
        --path-rename store/store.go:libnetwork/internal/kvstore/kvstore.go \
        --path-rename store/boltdb/:libnetwork/internal/kvstore/boltdb/
    
    
    # go to the target github.com/moby/moby repository
    cd ~/projects/docker
    
    # create a branch to work with
    git checkout -b integrate_libkv
    
    # add the temporary repository as an upstream and make sure it's up-to-date
    git remote add temp_libkv ~/projects/temp_libkv
    git fetch temp_libkv
    
    # merge the upstream code, rewriting "pkg/symlink" to "symlink"
    git merge --allow-unrelated-histories --signoff -S temp_libkv/migrate_libkv

See individual commits for details

**- What I did**

**- How I did it**

**- How to verify it**

**- Description for the changelog**

**- A picture of a cute animal (not mandatory but encouraged)**

### CODE REVIEW

**Comment by thaJeztah** [comment]
Ah, that was to be expected; probably needs to temporarily disable the DCO check;

```
These commits do not have a proper 'Signed-off-by:' marker:
 - 3fa22634a617e2c52d2c5f061826e5107e27985f
 - 9b11053e9147884c43c9a9d8ebfcd7bb9470e8b5
```

File: .github/workflows/.dco.yml [hunk 1]
```diff
@@ -35,14 +35,14 @@ jobs:
...
+#        name: Validate
+#        run: |
+#          docker run --rm \
+#            -v "$(pwd):/workspace" \
+#            -e VALIDATE_REPO \
+#            -e VALIDATE_BRANCH \
+#            alpine:${{ env.ALPINE_VERSION }} sh -c 'apk add --no-cache -q bash git openssh-client && git config --system --add safe.directory /workspace && cd /workspace && hack/validate/dco'
+#        env:
+#          VALIDATE_REPO: ${{ github.server_url }}/${{ github.repository }}.git
+#          VALIDATE_BRANCH: ${{ steps.base-ref.outputs.result }}
```
Comment (thaJeztah) [inline .github/workflows/.dco.yml:48]: Temporarily disabled the DCO check here (this must be reverted after merging, or we can drop this commit before merging (once CI passes))

**Comment by thaJeztah** [comment]
Ah, linter picked up some things;

```
libnetwork/kvstore/boltdb/boltdb.go:452:28: unnecessary conversion (unconvert)
			_ = bucket.Delete([]byte(key))
			                        ^
libnetwork/kvstore/boltdb/boltdb.go:425:2: S1023: redundant `return` statement (gosimple)
	return
	^
```

**Comment by thaJeztah** [comment]
Discussing in the moby sync call; I'll update this to move the package in to libnetwork/internal, to prevent it being used outside of libnetwork.

**Review by corhere — APPROVED** [review]

**Review by akerouanton — APPROVED** [review]
LGTM

File: .github/workflows/.dco.yml (hunk 1 above)
Comment (thaJeztah) [reply .github/workflows/.dco.yml:48]: - opened https://github.com/moby/moby/pull/45820 to re-enable it

**Pull request merged**